internal/controllers/image_controller.go

```diff
@@ -223,17 +223,44 @@ func (r *ImageReconciler) deleteImage(ctx context.Context, log logr.Logger, ioCt
 		return nil
 	}
 
-	if err := librbd.RemoveImage(ioCtx, ImageIDToRBDID(image.ID)); err != nil && !errors.Is(err, librbd.ErrNotFound) {
-		return fmt.Errorf("failed to remove rbd image: %w", err)
+	pools, imgs, err := listRbdImageChildren(ioCtx, ImageIDToRBDID(image.ID))
+	log.V(2).Info("Volume image references", "pools", pools, "rbd-images", imgs)
```

Comment:
I think this log message should come after the error check. Else we will log 0 pools and images if we have errors, which might not be correct.